### PR TITLE
Fixes client handling and module declaration

### LINKED ISSUE #20: Error the "id" argument must be of a type String
I had this code running before. Or I made a small change. I'm getting a strange error, I cant figure out what I'm doing wrong

**index.js**
```javascript
const Greenlock = require('greenlock');
const acmeDnsCloudflare = require('acme-dns-01-cloudflare');
const pkg = require('./package.json');

const TOKEN = "xxxx";
const EMAIL = "xxx@xxx"

const cloudflareDns01 = new acmeDnsCloudflare({
    token: TOKEN,
    verifyPropagation: true,
    verbose: true
});

console.log('dirname', __dirname);
const greenlock = Greenlock.create({
    staging: true,
    debug: true,
    maintainerEmail: EMAIL,
    packageRoot: "/home/xxx/cert_test",
    packageAgent: pkg.name + '/' + pkg.version,
    configDir: "/home/xxx/store",
    notify: function(event, details) {
        if ('error' === event) {
            // `details` is an error object in this case
            console.error(details);
        }
    }
});

greenlock.manager.defaults({
    agreeToTerms: true,
    subscriberEmail: EMAIL,
    packageRoot: __dirname,
    store: {
        module: "greenlock-store-fs",
        basePath: `/home/popcorn/WebstormProjects/webserver/cert_test/store/certs`
    },
    challenges: {
        "dns-01": cloudflareDns01
    }
});

greenlock.add({
    subject: "localh.app",
    altnames: ["localh.app", "*.localh.app"]
}).then(function(){
    console.log("SUCCESS");
}).catch(console.error);
```

Having an error:
```bash
Could not load 'undefined'
Did you install it?
        npm install --save undefined
TypeError [ERR_INVALID_ARG_TYPE]: The "id" argument must be of type string. Received undefined
    at new NodeError (node:internal/errors:259:15)
    at validateString (node:internal/validators:123:11)
    at Module.require (node:internal/modules/cjs/loader:965:3)
    at require (node:internal/modules/cjs/helpers:88:18)
    at Object.P._loadHelper (/home/xxx/cert_test/node_modules/@root/greenlock/plugins.js:29:32)
    at Object.P._loadChallenge (/home/popcorn/xxx/cert_test/node_modules/@root/greenlock/plugins.js:18:14)
    at /home/popcorn/xxx/cert_test/node_modules/@root/greenlock/greenlock.js:456:26
    at Array.map (<anonymous>)
    at Object.greenlock._order (/home/popcorn/xxx/cert_test/node_modules/@root/greenlock/greenlock.js:455:41)
    at async Object.greenlock._renew (/home/popcorn/xxx/cert_test/node_modules/@root/greenlock/greenlock.js:335:9) {
```

### PR DESCRIPTION
**Pull request by ajuanjojjj**
Fixes #20

### CODE REVIEW

**Comment by ajuanjojjj** [comment]
I'm not super familiar with all the in's and out's of the module, so i'm not sure if i'm missing something wrong with these changes, feel free to point it out!

**Review by Cherry — COMMENTED** [review]
Thanks so much for the PR! I'll review and test this next week.

File: index.js [hunk 1]
```diff
@@ -58,12 +58,19 @@ function delay(ms){
...
-			key: options.key,
-			token: options.token
+		this.cfClient = new cloudflare({
+			email: options.client?.email || options.email,
+			key: options.client?.key || options.key,
+			token: options.client?.token || options.token
 		});
+		this.client = {
+
+			email: options.cfClient?.email || options.client?.email || options.email,
```
Comment (Cherry) [inline index.js:70]: Will `options.cfClient` ever be set? Perhaps this is meant to be `this.cfClient`? Also, what's the need for setting `this.client` at all? Is this something `greenlock` etc. relies on?

File: index.js [hunk 2]
```diff
@@ -58,12 +58,19 @@ function delay(ms){
...
+			key: options.cfClient?.key || options.client?.key || options.key,
+			token: options.cfClient?.token || options.client?.token || options.token
+		}
```
Comment (Cherry) [inline index.js:73]: If you fix up this minor issue flagged by ESLint, that would be great!

Comment (ajuanjojjj) [reply index.js]: Done!

File: index.js (hunk 1 above)
Comment (ajuanjojjj) [reply index.js]: It's the constructor, so it wouldn't be set in the `this` scope yet. However, its true that it'll probably wont ever be set in `options` (Same as the one below), that was just leftover code from when i didn't fully understand the issue

Comment (ajuanjojjj) [reply index.js]: As for why this.client is being set, thats the client greenlock seems to be storing in the options file; However, the cloudflare prototype seems to have problems being stringified to json, and it wasn't being stored. Changing it to a simple object without prototype solved it on my end

**Review by ajuanjojjj — COMMENTED** [review]
Solved the missing ; and explained the this.client and this.cfClient changes

**Review by Cherry — APPROVED** [review]

**Comment by Cherry** [comment]
Thanks. All looks good - merging now.

**Pull request merged**